File: configlibrary/src/main/java/com/shmedo/configlibrary/iot/cmd/entity/adme/AdmeLockedRotorDetectionEntity.java
package com.shmedo.configlibrary.iot.cmd.entity.adme;

import com.shmedo.configlibrary.ble.interfaces.Validater;

import java.lang.reflect.Field;

/**
 * 创建者:   gonghe <br/>
 * 创建时间:  3/2/21 <br/>
 * 描述：     生成ADME 电机运动堵转缓停参数拼接指令
 */
public class AdmeLockedRotorDetectionEntity implements Validater {
    private String lowtbtss;//下放堵转缓停（0:关闭，1:开启）
    private String numpput;//单位时间脉冲数
    private String pdajtime;//脉冲检测判断时间
    private String detintiona;//堵转检测区间起始值
    private String detintionb;//堵转检测区间终值
    private String lowtorblothr;//下放力矩堵转阈值
    private String lowtordetime;//下放力矩检测判断时间
    private String lowsusrana;//下放缓停区间起始值
    private String lowsusranb;//下放缓停区间终值
    private String uptbtss;//上拉堵转缓停（0:关闭，1:开启）
    private String uptorblothr;//上拉力矩堵转阈值
    private String uptordetime;//上拉力矩检测判断时间
    private String upsusrana;//上拉缓停区间起始值
    private String upsusranb;//上拉缓停区间终值

    public void setLowtbtss(String lowtbtss) {
        this.lowtbtss = lowtbtss;
    }

    public void setNumpput(String numpput) {
        this.numpput = numpput;
    }

    public void setPdajtime(String pdajtime) {
        this.pdajtime = pdajtime;
    }

    public void setDetintiona(String detintiona) {
        this.detintiona = detintiona;
    }

    public void setDetintionb(String detintionb) {
        this.detintionb = detintionb;
    }

    public void setLowtorblothr(String lowtorblothr) {
        this.lowtorblothr = lowtorblothr;
    }

    public void setLowtordetime(String lowtordetime) {
        this.lowtordetime = lowtordetime;
    }

    public void setLowsusrana(String lowsusrana) {
        this.lowsusrana = lowsusrana;
    }

    public void setLowsusranb(String lowsusranb) {
        this.lowsusranb = lowsusranb;
    }

    public void setUptbtss(String uptbtss) {
        this.uptbtss = uptbtss;
    }

    public void setUptorblothr(String uptorblothr) {
        this.uptorblothr = uptorblothr;
    }

    public void setUptordetime(String uptordetime) {
        this.uptordetime = uptordetime;
    }

    public void setUpsusrana(String upsusrana) {
        this.upsusrana = upsusrana;
    }

    public void setUpsusranb(String upsusranb) {
        this.upsusranb = upsusranb;
    }

    @Override
    public void validate() {

    }

    @Override
    public String toString() {
        StringBuilder stringBuilder = new StringBuilder();
        try {
            for (Field f : getClass().getDeclaredFields()) {
                Object value = f.get(this);
                if (value != null && !value.equals("NullKey")) {
                    stringBuilder.append(f.getName());
                    stringBuilder.append("=");
                    stringBuilder.append(value);
                    stringBuilder.append("&");
                }
            }
        } catch (IllegalAccessException e) {
            e.printStackTrace();
        }
        if (stringBuilder.toString().endsWith("&")) {
            stringBuilder.delete(stringBuilder.length() - 1, stringBuilder.length());
        }
        return stringBuilder.toString();
    }
}

File: lib_device_base/src/main/java/com/shmedo/lib/device/base/iot_cmd/entity/adme/AdmeLockedRotorDetectionEntity.kt
package com.shmedo.lib.device.base.iot_cmd.entity.adme

import com.shmedo.lib.device.base.Validater

/**
 * 创建者:   gonghe <br></br>
 * 创建时间:  3/2/21 <br></br>
 * 描述：     生成ADME 电机运动堵转缓停参数拼接指令
 */
class AdmeLockedRotorDetectionEntity : Validater {
    var lowtbtss //下放堵转缓停（0:关闭，1:开启）
            : String? = null
    var numpput //单位时间脉冲数
            : String? = null
    var pdajtime //脉冲检测判断时间
            : String? = null
    var detintiona //堵转检测区间起始值(匀速阶段)
            : String? = null
    var detintionb //堵转检测区间终值(匀速阶段)
            : String? = null
    var lowtorblothr //下放力矩堵转阈值
            : String? = null
    var lowtordetime //下放力矩检测判断时间
            : String? = null
    var lowsusrana //下放缓停区间起始值(减速阶段)
            : String? = null
    var lowsusranb //下放缓起区间终值(加速阶段)
            : String? = null
    var uptbtss //上拉堵转缓停（0:关闭，1:开启）
            : String? = null
    var uptorblothr //上拉力矩堵转阈值
            : String? = null
    var uptordetime //上拉力矩检测判断时间
            : String? = null
    var upsusrana //上拉缓停区间起始值(减速阶段)
            : String? = null
    var upsusranb //上拉缓起区间终值(加速阶段)
            : String? = null

    override fun validate() {}
    override fun toString(): String {
        val stringBuilder = StringBuilder()
        try {
            for (f in javaClass.declaredFields) {
                val value = f[this]
                if (value != null && value != "NullKey") {
                    stringBuilder.append(f.name)
                    stringBuilder.append("=")
                    stringBuilder.append(value)
                    stringBuilder.append("&")
                }
            }
        } catch (e: IllegalAccessException) {
            e.printStackTrace()
        }
        if (stringBuilder.toString().endsWith("&")) {
            stringBuilder.delete(stringBuilder.length - 1, stringBuilder.length)
        }
        return stringBuilder.toString()
    }
}